CONTRIBUTING.md

```diff
@@ -1,39 +1,37 @@
+## GitHub Flow
+We follow the GitHub Flow for making code changes. This means that all code modifications should be proposed through pull requests. Pull requests provide a structured and collaborative way to review and discuss code changes.
 
-1. Fork the repo and create your branch from `master`.
-2. If you've added code that should be tested, add tests.
-3. If you've changed APIs, update the documentation.
-4. Ensure the CI passes.
-5. Issue that pull request!
+If you'd like to contribute, please follow these steps:
 
-## Any contributions you make will be under the MIT Software License
-In short, when you submit code changes, your submissions are understood to be under the same [MIT License](http://choosealicense.com/licenses/mit/) that covers the project. Feel free to contact the maintainers if that's a concern.
+1. Select an open issue to work on or create a new issue if none exists.
+2. Fork the repository and create a branch from the `master` branch.
+   - Use the issue number as the branch name, e.g., GD-111.
+   - Assign the issue to yourself and set its status to "In Progress."
+3. If you have made changes to the code that should be tested, please include appropriate tests.
+4. If you have modified any APIs, ensure that the documentation is updated accordingly.
+5. Create a pull request and provide information in the "Why" and "What" sections:
+   - Link the pull request to the corresponding issue.
+   - Assign the pull request to yourself.
+   - Make sure each pull request is associated with only one issue.
+   - If the pull request is still in progress, mark it as a draft.
+   - Ensure that the continuous integration (CI) process passes successfully.
+6. Submit the pull request!
 
-## Report bugs using Github's
-We use GitHub issues to track public bugs. Report a bug by [opening a new issue](https://github.com/MikeSchulze/gdUnit4/issues/new?assignees=MikeSchulze&labels=bug%2C+task&template=bug_report.md&title=); it's that easy!
+## License
+By contributing to this project, you agree that your contributions will be licensed under the same [MIT License](https://github.com/MikeSchulze/gdUnit4/blob/master/LICENSE) that covers the project. If you have any concerns, please reach out to the maintainers.
 
-Write bug reports with detail, background, and sample code
+## Reporting Bugs
+If you encounter any bugs or issues, please use GitHub's issue tracking system. You can report a bug by [opening a new issue](https://github.com/MikeSchulze/gdUnit4/issues/new?assignees=MikeSchulze&labels=bug%2C+task&template=bug_report.md&title=). When submitting a bug report, please provide detailed information, including the steps to reproduce the issue, relevant background information, and sample code if possible.
 
-## Use a Consistent Coding Style
-I'm again borrowing these from 
+## Coding Style
```

Comment:
Can't this go into one of the PR steps? Step 5 maybe?

- Ensure your code follows [GDScript](https://docs.godotengine.org/en/stable/tutorials/scripting/gdscript/gdscript_styleguide.html) or [C#](https://docs.microsoft.com/en-us/dotnet/csharp/fundamentals/coding-style/coding-conventions)  Coding Conventions